datadog/resource_datadog_dashboard.go

```diff
@@ -5657,6 +6088,154 @@ func buildDatadogGroupBySort(sort map[string]interface{}) *datadogV1.LogQueryDef
+	for i, query := range datadogQueries {
+		terraformQuery := map[string]interface{}{}
+		if query.TimeSeriesFormulaAndFunctionEventQueryDefinition != nil {
+			if dataSource, ok := query.TimeSeriesFormulaAndFunctionEventQueryDefinition.GetDataSourceOk(); ok {
+				terraformQuery["data_source"] = dataSource
+			}
+			if name, ok := query.TimeSeriesFormulaAndFunctionEventQueryDefinition.GetNameOk(); ok {
+				terraformQuery["name"] = name
+			}
+			if indexes, ok := query.TimeSeriesFormulaAndFunctionEventQueryDefinition.GetIndexesOk(); ok {
+				terraformQuery["indexes"] = indexes
+			}
+			if search, ok := query.TimeSeriesFormulaAndFunctionEventQueryDefinition.GetSearchOk(); ok {
+				if len(search.GetQuery()) > 0 {
+					terraformSearch := map[string]interface{}{}
+					terraformSearch["query"] = search.GetQuery()
+					terraformSearchList := []map[string]interface{}{terraformSearch}
+					terraformQuery["search"] = terraformSearchList
+				}
+			}
+			if compute, ok := query.TimeSeriesFormulaAndFunctionEventQueryDefinition.GetComputeOk(); ok {
+				terraformCompute := map[string]interface{}{}
+				if aggregation, ok := compute.GetAggregationOk(); ok {
+					terraformCompute["aggregation"] = aggregation
+				}
+				if interval, ok := compute.GetIntervalOk(); ok {
+					terraformCompute["interval"] = interval
+				}
+				if metric, ok := compute.GetMetricOk(); ok {
+					terraformCompute["metric"] = metric
+				}
+				terraformComputeList := []map[string]interface{}{terraformCompute}
+				terraformQuery["compute"] = terraformComputeList
+			}
+			if v, ok := query.TimeSeriesFormulaAndFunctionEventQueryDefinition.GetGroupByOk(); ok {
+				terraformGroupBys := make([]map[string]interface{}, len(*v))
+				for i, groupBy := range *v {
+					// Facet
+					terraformGroupBy := map[string]interface{}{
+						"facet": groupBy.GetFacet(),
+					}
+					// Limit
+					if v, ok := groupBy.GetLimitOk(); ok {
+						terraformGroupBy["limit"] = *v
+					}
+					// Sort
+					if v, ok := groupBy.GetSortOk(); ok {
+						terraformSort := map[string]interface{}{}
+						if metric, ok := v.GetMetricOk(); ok {
+							terraformSort["metric"] = metric
+						}
+						if order, ok := v.GetOrderOk(); ok {
+							terraformSort["order"] = order
+						}
+						if aggregation, ok := v.GetAggregationOk(); ok {
+							terraformSort["aggregation"] = aggregation
+						}
+						terraformGroupBy["sort"] = []map[string]interface{}{terraformSort}
+					}
+					terraformGroupBys[i] = terraformGroupBy
+				}
+				terraformQuery["group_by"] = &terraformGroupBys
+			}
+			terraformQueries := []map[string]interface{}{terraformQuery}
+			terraformEventQuery := map[string]interface{}{}
+			terraformEventQuery["event_query"] = terraformQueries
+			queries[i] = terraformEventQuery
+		}
+		if query.TimeSeriesFormulaAndFunctionMetricQueryDefinition != nil {
+			if dataSource, ok := query.TimeSeriesFormulaAndFunctionMetricQueryDefinition.GetDataSourceOk(); ok {
```

Comment:
Same nit here on storing this in a variable

Reply:
Addressed in commit [5c22cd7](https://github.com/DataDog/terraform-provider-datadog/pull/892/commits/5c22cd771533e25ac58a9e87bc9da024672d34f3).